tests/Transforms/shape_inference/shape_inference.mlir

```diff
@@ -0,0 +1,50 @@
+// RUN: heir-opt --shape-inference  %s | FileCheck %s
+
+//CHECK-LABEL: func @no_inference
+//CHECK-SAME:(%[[X:.*]]: tensor<42xi16>, %[[Y:.*]]: tensor<42xi16>) -> tensor<42xi16>
+func.func @no_inference(%x: tensor<?xi16> {shape.shape=[42]} , %y: tensor<?xi16> {shape.shape=[42]}) -> (tensor<?xi16>) {
+    //CHECK-NEXT: %[[ADD:.*]] = arith.addi %[[X]], %[[Y]] : tensor<42xi16>
+    %0 = arith.addi %x, %y : tensor<?xi16>
+    //CHECK-NEXT: return %[[ADD]] : tensor<42xi16>
+    func.return %0 : tensor<?xi16>
+}
+
+//CHECK-LABEL: func @one_dimensional
+//CHECK-SAME:(%[[X:.*]]: tensor<42xi16>, %[[Y:.*]]: tensor<42xi16>) -> tensor<42xi16>
+func.func @one_dimensional(%x: tensor<?xi16> {shape.shape=[42]} , %y: tensor<?xi16> {shape.shape=[42]}) -> (tensor<?xi16>) {
+    //CHECK-NEXT: %[[ADD:.*]] = arith.addi %[[X]], %[[Y]] : tensor<42xi16>
+    %0 = arith.addi %x, %y : tensor<?xi16>
+    //CHECK-NEXT: return %[[ADD]] : tensor<42xi16>
+    func.return %0 : tensor<?xi16>
+}
+
+//CHECK-LABEL: func @two_dimensional
```

Comment:
here and elsewhere, just use `// CHECK:`. The regex in the pre-commit should have caught this, but there's no space before `C` after `//`, so I'm not even sure these assertions are being triggered by `FileCheck` (I thought the space was mandatory, but maybe it's not?)

Reply:
Done @CHECK-LABEL, the space isn't mandatory as far as I can tell (had plenty of such CHECKs trigger during debugging 😉) and we've already got a bunch of other tests without spaces, so the pre-commit regex might need updating :)